src/pybind/mgr/cephadm/services/jaeger.py

```diff
@@ -20,13 +20,16 @@ class JaegerAgentService(CephadmService):
     def prepare_create(self, daemon_spec: CephadmDaemonDeploySpec) -> CephadmDaemonDeploySpec:
         assert self.TYPE == daemon_spec.daemon_type
         collectors = []
+        deps: List[str] = []
         for dd in self.mgr.cache.get_daemons_by_type(JaegerCollectorService.TYPE):
             # scrape jaeger-collector nodes
             assert dd.hostname is not None
             port = dd.ports[0] if dd.ports else JaegerCollectorService.DEFAULT_SERVICE_PORT
             url = build_url(host=dd.hostname, port=port).lstrip('/')
             collectors.append(url)
+            deps.append(dd.name())
```

Comment:
this original version just adding the daemon names as deps doesn't seem to have worked. I think it's because what's actually being set up relies on the ports of the jaeger-collectors which briefly may not be known even when the jaeger-collector has been placed since we pick up the ports in the next daemon refresh. So if the jaeger-collector and jaeger agent are deployed in the same serve loop iteration it can end up not reoconfiguring once it finds out the ports later.